quix-frontend/service/src/modules/event-sourcing/quix-event-bus.spec.ts

```diff
@@ -519,6 +519,74 @@ describe('event sourcing', () => {
       expect(notebook.notes![0].textContent).toBe('select foo from bar');
     });
 
+    describe('note update', () => {
+      beforeEach(async () => {
+        await driver.emitAsUser(eventBus, [
+          createNotebookAction,
+          addNoteAction,
+        ]);
+      });
+
+      it.each([
+        ['content', () => NoteActions.updateContent(note.id, 'content')],
+        [
+          'richContent',
+          () => NoteActions.updateContent(note.id, '\n', {rich: 'content'}),
+        ],
+        ['name', () => NoteActions.updateName(note.id, 'name')],
+      ])(
+        'should not update %s when there are no changes',
+        async (_text, createAction) => {
+          const action = createAction();
+
+          await driver.emitAsUser(eventBus, [action]);
+          const firstNote = await driver.getNote(note.id);
+
+          await new Promise(resolve => setTimeout(resolve, 1000));
+
+          await driver.emitAsUser(eventBus, [action]);
+          const secondNote = await driver.getNote(note.id);
+
+          expect(firstNote?.dateUpdated).toBe(secondNote?.dateUpdated);
```

Comment:
you are using `it.each`, so you should define your own error to understand exactly which case is failed (if it failed)